mods/default/craftitems.lua

```diff
@@ -83,18 +137,19 @@ minetest.register_on_player_receive_fields(function(player, formname, fields)
 				stack:set_count(count - 1)
 				new_stack = ItemStack("default:book_written")
 			end
-		else
-			data = stack:get_meta():to_table().fields
 		end
 
-		if data and data.owner and data.owner ~= player:get_player_name() then
+		if data.owner ~= player_name and title ~= "" and text ~= "" then
 			return
 		end
 
 		if not data then data = {} end
 		data.title = fields.title:sub(1, max_title_size)
 		data.owner = player:get_player_name()
 		local short_title = data.title
+		if short_title == "" then
+			short_title = S("Book")
+		end
```

Comment:
```suggestion
```
This is potentially not a good idea due to the shortening logic below and also looks weird when the item says `"Book" by singleplayer` when it's actually unnamed, so removing this.